packages/react-ui/src/components/molecules/UploadField.js

```diff
@@ -0,0 +1,198 @@
+      setFilesText(props.files[0].name);
+    } else {
+      setFilesText(`${props.files.length} files selected`);
+    }
+  }, [props.files]);
+
+  const handleBrowse = () => {
+    uploadInputRef.current?.click();
+  };
+
+  const handleDragOver = (event) => {
+    event.preventDefault();
+    setDragOver(true);
+  };
+
+  const handleDragLeave = (event) => {
+    event.preventDefault();
+    setDragOver(false);
+  };
+
+  const handleDrop = (event) => {
+    event.preventDefault();
+    setDragOver(false);
+
+    const items = event.dataTransfer.items;
+    const newFiles = getAllFiles(items);
+    props.onChange(newFiles);
+  };
+
+  const getAllFiles = (items) => {
+    const newFiles = [];
+    for (let item of items) {
+      if (item.kind === 'file') {
+        const file = item.getAsFile();
+        newFiles.push(file);
+      }
+    }
+
+    return newFiles;
+  };
+
+  const handleFiles = (event) => {
+    const newFiles = Array.from(event.target.files);
+    props.onChange(newFiles);
+  };
+
+  const handleReset = (event) => {
+    event.stopPropagation();
+
+    setFilesText('');
+  };
+
+  const dragPlaceholderText = dragOver ? props.dragPlaceholder : props.placeholder;
+
+  return (
+    <>
+      <TextField
+        {...props}
+        ref={textFieldRef}
+        variant={props.variant}
+        placeholder={dragPlaceholderText}
+        value={filesText}
+        label={props.label}
+        helperText={props.helperText}
+        error={props.error}
+        className={`${classes.uploadField} ${dragOver && classes.focused}`}
+        InputProps={{
+          onDragOver: handleDragOver,
+          onDragLeave: handleDragLeave,
+          onDrop: handleDrop,
+          onClick: handleBrowse,
+          readOnly: true,
+          endAdornment: (
+            <InputAdornment position='end'>
+              {!filesText ? (
+                <Button
+                  size='small'
+                  variant='text'
+                  color={props.error ? 'default' : 'primary'}
+                  disabled={!!dragOver}
+                  className={classes.button}
+                >
+                  {props.buttonText}
+                </Button>
+              ) : (
+                <IconButton
+                  onClick={handleReset}
+                  aria-label='delete'
+                  disabled={!!dragOver}
+                  size='small'
+                >
+                  <Cancel />
+                </IconButton>
+              )}
+            </InputAdornment>
+          )
+        }}
+      />
+      <input
+        ref={uploadInputRef}
+        style={{ display: 'none' }}
+        type='file'
+        accept={props.accept}
+        multiple={props.multiple}
+        onChange={handleFiles}
+      />
+    </>
+  );
+}
+
+UploadField.defaultProps = {
+  placeholder: 'Drag and drop your file or click to browse',
+  dragPlaceholder: 'Drop your file here...',
+  buttonText: 'Browse',
+  accept: 'application/JSON',
```

Comment:
Is this comma separated, in the case of various types? Should use the opportunity to have accept as an array externally?

Reply:
I updated the prop, I think I got you right, if not feel free to modify anything. Thank you!